bot/exts/evergreen/wolfram.py

```diff
@@ -62,7 +62,8 @@ async def predicate(ctx: Context) -> bool:
             # if the invoked command is help we don't want to increase the ratelimits since it's not actually
             # invoking the command/making a request, so instead just check if the user/guild are on cooldown.
             guild_cooldown = not guildcd.get_bucket(ctx.message).get_tokens() == 0  # if guild is on cooldown
-            if not any(r.id in ignore for r in ctx.author.roles):  # check user bucket if user is not ignored
+            # check the message is in a guild, and check user bucket if user is not ignored
+            if ctx.guild and not any(r.id in ignore for r in ctx.author.roles):
```

Comment:
```suggestion
            if not (ctx.guild and any(r.id in ignore for r in ctx.author.roles)):
```

Reply:
I'm trying to check if the ctx is within the guild, wouldn't this change would invert that?

Reply:
I understood the `ignore` roles part wrong, my bad.